app/assets/javascripts/discourse/initializers/register-service-worker.js.es6

```diff
@@ -6,9 +6,10 @@ export default {
           (location.hostname === "localhost");
 
     const isSupported= isSecured && ('serviceWorker' in navigator);
+    const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
```

Comment:
We might already have that User Agent sniffing code somewhere in a lib. Might be worth checking out.

Reply:
I found one for sniffing out Apple, but no dice checking for Safari specifically.

For reference, "Safari" also appears in macOS' Chrome, which is why the regex checks for the strings "chrome" or "android" not appearing as well.